pennylane/debugging/operation_checker.py

```diff
@@ -0,0 +1,833 @@
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``terms``. ``None`` if no terms are defined via ``terms``
+        or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    terms = wrap_op_method(instance, "terms", TermsUndefinedError)()
+    if terms is None or isinstance(terms, Exception):
+        return terms
+
+    return np.sum([c * qml.matrix(op) for c, op in zip(*terms)], axis=0)
+
+
+def matrix_from_decomposition(op, par, wires):
+    r"""Get the matrix of an operation, using its ``decomposition``.
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``expand``. ``None`` if no decomposition is defined via ``expand``
+        or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    tape = wrap_op_method(instance, "expand", DecompositionUndefinedError)()
+    return qml.matrix(tape, wire_order=wires) if isinstance(tape, qml.tape.QuantumTape) else tape
+
+
+def matrix_from_single_qubit_rot_angles(op, par, wires):
+    r"""Get the matrix of an operation, using its ``single_qubit_rot_angles``.
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``single_qubit_rot_angles``. ``None`` if no rotation angles are
+        defined via ``single_qubit_rot_angles`` or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    try:
+        angles = instance.single_qubit_rot_angles()
+    except (AttributeError, NotImplementedError):
+        return None
+    with qml.tape.QuantumTape() as tape:
+        qml.RZ(angles[0], wires=wires)
+        qml.RY(angles[1], wires=wires)
+        qml.RZ(angles[2], wires=wires)
+    return qml.matrix(tape)
+
+
+def matrix_from_generator(op, par, wires):
+    r"""Get the matrix of an operation, using its ``generator``.
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``generator``. ``None`` if no generator is defined via ``generator``
+        or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    gen = wrap_op_method(instance, "generator", GeneratorUndefinedError)()
```

Comment:
Does it make sense to just rely on `qml.generator()` here? Above, you are using `qml.matrix`?